pkg/imgpkg/imageset/tar_image_set.go

```diff
@@ -27,21 +29,64 @@ func NewTarImageSet(imageSet ImageSet, concurrency int, ui goui.UI) TarImageSet
 }
 
 // Export Creates a Tar with the provided Images
-func (i TarImageSet) Export(foundImages *UnprocessedImageRefs, outputPath string, registry registry.ImagesReaderWriter, imageLayerWriterCheck imagetar.ImageLayerWriterFilter) (*imagedesc.ImageRefDescriptors, error) {
+func (i TarImageSet) Export(foundImages *UnprocessedImageRefs, outputPath string, registry registry.ImagesReaderWriter, imageLayerWriterCheck imagetar.ImageLayerWriterFilter, resume bool) (d *imagedesc.ImageRefDescriptors, err error) {
 	ids, err := i.imageSet.Export(foundImages, registry)
 	if err != nil {
 		return nil, err
 	}
 
-	outputFile, err := os.Create(outputPath)
+	var outputFile *os.File
+	var layers []v1.Layer
+	var tmpFile *os.File
+	if resume {
+		tmpFile, err = ioutil.TempFile("", "tmp-extract")
+		if err != nil {
+			return nil, fmt.Errorf("Creating tmp folder: %s", err)
+		}
+		defer os.Remove(tmpFile.Name())
+
+		outputFile, err = os.Open(outputPath)
+		if err != nil {
+			return nil, fmt.Errorf("Opening file '%s': %s", outputPath, err)
+		}
```

Comment:
it seems that first invocation of imgpkg would not be able to contain --resume, since output path might not exist. since this functionality would typically be used in a loop to retry i would recommend to allow absense of a file be allowed when --resume is specified.

Reply:
sounds good, updated the PR to continue like if --resume was not provided